_posts/2015-10-20-jenkins-followup.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: "Followup: Automating Jenkins"
+date: "2015-10-20 09:00:00"
+icon: wrench
+categories: tech-stories
+authors: ["Dirk Jablonski"]
+---
+
+Some weeks ago, we wrote on [Infrstructure as Code: automating Jenkins](https://developer.epages.com/blog/2015/06/25/infrastructure-as-code.html).
+In that post, we described some open ends, which still needed to be solved in order to satisfy our expectations.
+In this followup post, I will explain how we advanced on our way to have a fully automated continuous integration server, this time coming pretty close to an *[immutable server](http://martinfowler.com/bliki/ImmutableServer.html)*.
+
+## Remaining shortcomings
+
+The main problem we had earlier was the central configuration of the Jenkins master, which is extended by lots of plugins.
+As a workaround, we created the full `config.xml` in advance and just copied it over to Jenkins.
+In addition, we were pretty unhappy with not being able to avoid local modifications, which were still leaving permanent traces behind, even when being overwritten by a new run of Ansible.
+
+## Dockerize!
+
+{% image docker-logo-h.png %}{% endimage %}
+
+So, let's get started on our journey towards immutable Jenkins master!
+The main step to achieve this goal was our decision to build our further steps on [Docker](https://www.docker.com/).
```

Comment:
The main step to achieve this goal was based on the decision to build further steps on...